trace/tracestate_test.go

```diff
@@ -443,14 +443,21 @@ func TestTraceStateInsert(t *testing.T) {
 			err:        errInvalidKey,
 		},
 		{
-			name:       "too many entries",
+			name:       "drop the right-most member(oldest) in queue",
 			tracestate: maxMembers,
 			key:        "keyx",
 			value:      "valx",
-			expected:   maxMembers,
-			err:        errMemberNumber,
-		},
-	}
+			expected: func() TraceState {
```

Comment:
I'm not finding this method super as to what it's doing. How about something like this:

```go
expected: func() TraceState {
	// Prepend the new element and remove the oldest one, which is over capacity.
	return TraceState{
		list: append(
			[]member{{Key: "keyx", Value: "valx"}},
			maxMembers.list[:len(maxMembers.list)-1]...,
		),
	}
}(),
```